pennylane/transforms/core/transform.py

```diff
@@ -177,15 +178,44 @@ def qnode_circuit(a):
             )
 
     # 3: CHeck the classical co-transform
-    if classical_cotransform is not None:
-        if not callable(classical_cotransform):
-            raise TransformError("The classical co-transform must be a valid Python function.")
+    if classical_cotransform is not None and not callable(classical_cotransform):
+        raise TransformError("The classical co-transform must be a valid Python function.")
 
-    dispatcher = TransformDispatcher(
+    return TransformDispatcher(
         quantum_transform,
         expand_transform=expand_transform,
         classical_cotransform=classical_cotransform,
         is_informative=is_informative,
         final_transform=final_transform,
     )
-    return dispatcher
+
+
+def null_postprocessing(results):
+    """A postprocessing function with null behavior."""
+    return results[0]
+
+
+def expand_fn_transform(
```

Comment:
I'll add the `*args, **kwargs`.

Right now, I mostly see this as a patch for things that we can't easily move to the new system, rather than a recommended tool.  

I'm ok with either the name `single_tape_transform` or `expand_fn_transform`. Any opinions @trbromley ?

Reply:
Good point about `*args, **kwargs`.

The purpose behind this was more for patching pre-existing code we can't easily switch.  I'm not sure we should particularly recommend it or use it places where we can use the alternative.

Reply:
Note this is something external users writing transforms will want/have wanted as well, so definitely something worth considering from a user POV.

For a researcher, the most ideal UI would simply be 

```python
@qml.transform
def my_transform(tape):
    new_ops = some_stuff(tape)
    new_tape = type(tape)(new_ops, tape.measurements, shots=tape.shots)
    return new_tape
```

That is, if no post-processing function/list of tapes is provided, it is treated as a single tape transform . It's a tradeoff --- more complex logic which differs depending on the return form, at the advantage of a single decorator the user needs to know about.

Out of scope for this PR, but something I wanted to mention for future consideration (cc @trbromley).

Reply:
For now, what name do we want for this function and what import level do we make it accessible at?

Reply:
I'm removing this function from the transforms folder and making it private to `construct_batch.py`